packages/datagateway-download/src/downloadStatus/adminDownloadStatusTable.component.tsx

```diff
@@ -115,110 +110,72 @@ const AdminDownloadStatusTable: React.FC = () => {
-        downloadApiUrl: settings.downloadApiUrl,
-      },
-      queryOffset
-    ).then((downloads) => {
-      const formattedDownloads = formatDownloads(downloads);
-      setData([...formattedDownloads]);
-    });
-  }, [
-    buildQueryOffset,
-    formatDownloads,
-    settings.downloadApiUrl,
-    settings.facilityName,
-  ]);
+  const {
+    data,
+    isLoading,
+    isFetched,
+    fetchNextPage,
+    refetch,
+  } = useAdminDownloads({
+    initialQueryOffset: `${buildQueryOffset()} LIMIT 0, 50`,
+  });
 
   const fetchMoreData = useCallback(
-    (offsetParams: IndexRange) => {
-      let queryOffset = buildQueryOffset();
-      queryOffset += ` LIMIT ${offsetParams.startIndex}, ${
-        offsetParams.stopIndex - offsetParams.startIndex + 1
-      }`;
-
-      setDataLoaded(false);
-      return fetchAdminDownloads(
-        {
-          facilityName: settings.facilityName,
-          downloadApiUrl: settings.downloadApiUrl,
-        },
-        queryOffset
-      ).then((downloads) => {
-        const formattedDownloads = formatDownloads(downloads);
-        setData([...data, ...formattedDownloads]);
-        setDataLoaded(true);
-      });
-    },
-    [
-      buildQueryOffset,
-      data,
-      formatDownloads,
-      settings.downloadApiUrl,
-      settings.facilityName,
-    ]
+    (offsetParams: IndexRange) =>
+      fetchNextPage({
+        pageParam: `${buildQueryOffset()} LIMIT ${offsetParams.startIndex}, ${
+          offsetParams.stopIndex - offsetParams.startIndex + 1
+        }`,
+      }),
+    [buildQueryOffset, fetchNextPage]
   );
 
+  const refreshTable = useCallback(async () => {
+    await refetch();
+    setRefreshDownloads(false);
+  }, [refetch]);
+
   React.useEffect(() => {
-    // Clear the current contents, this will make sure
-    // there is visually a refresh of the table
-    setData([]);
-
-    if (dgDownloadElement) {
-      setDataLoaded(false);
-      fetchInitialData();
-      setDataLoaded(true);
+    if (refreshDownloads && isFetched) {
+      // user requested to refresh table
+      refreshTable();
     }
-  }, [
-    settings.facilityName,
-    settings.downloadApiUrl,
-    dgDownloadElement,
-    filters,
-    sort,
-    fetchInitialData,
-  ]);
+  }, [isFetched, refreshDownloads, refreshTable]);
 
   React.useEffect(() => {
-    if (refreshDownloads) {
-      setData([]);
-      setDataLoaded(false);
-      setRefreshDownloads(false);
-      fetchInitialData();
+    if (isFetched) {
       setLastChecked(new Date().toLocaleString());
-      setDataLoaded(true);
     }
-  }, [fetchInitialData, refreshDownloads]);
+  }, [isFetched]);
+
+  React.useEffect(() => {
+    // refetch table whenever filter changes
+    // debounced to avoid excessive calls
+    refetchTimeout.current = setTimeout(() => {
+      refetch();
+    }, 200);
+
+    return () => {
+      const prevTimeout = refetchTimeout.current;
+      if (prevTimeout) clearTimeout();
+    };
+  }, [refetch, filters]);
```

Comment:
This code shouldn't be necessary - as `TextFilter` already debounces it's input - see https://github.com/ral-facilities/datagateway/blob/6292c13ff7b41a5f31abad669e97cf29551b3852/packages/datagateway-common/src/table/columnFilters/textColumnFilter.component.tsx#L21.

Reply:
Interesting - the test times out without the debounce, I'll investigate further.